src/components/file-handler/FileHandler.jsx

```diff
@@ -0,0 +1,167 @@
+// @flow strict
+
+import React from 'react';
+import type {Node} from 'react';
+import classNames from 'classnames';
+
+import Text from '../text/Text';
+import Link from '../text/Link';
+import Icon from '../icons/Icon';
+import Spinner from '../spinner/Spinner';
+
+import type {IconTypeType} from '../icons/Icon';
+
+export type FileHandlerColorType = 'gray' | 'mono';
+
+export const COLORS_MAP = {
+  gray: 'gray-secondary-light',
+  mono: 'white',
+};
+
+export const FILE_HANDLER_COLORS_SET = {
+  GRAY: 'gray',
+  MONO: 'mono',
+};
+
+type PropsType = $ReadOnly<{
+  /**
+   * Specify color of the background for FileHandler
+   * @example <FileHandler color="mono">
+   *            text
+   *          </FileHandler>
+   * @default gray
+   */
+  color: FileHandlerColorType,
+  /**
+   * Specify iconType to display SG icon as the image inside FileHandler
+   * @example <FileHandler iconType="answer">
+   *            text
+   *          </FileHandler>
+   * @default attachment
+   */
+  iconType?: IconTypeType,
+  /**
+   * Specify thumbnailSrc of the image inside FileHandler
+   * @example <FileHandler thumbnailSrc="https://source.unsplash.com/64x64/?bird">
+   *            text
+   *          </FileHandler>
+   */
+  thumbnailSrc?: string,
+  /**
+   * Specify src of the file to display it in the new tab when link is clicked
+   * @example <FileHandler src="https://source.unsplash.com/64x64/?bird">
+   *            text
+   *          </FileHandler>
+   */
+  src?: string,
+  /**
+   * Optional boolean for loading state of FileHandler
+   * @example <FileHandler loading>
+   *            text
+   *          </FileHandler>
+   * @default false
+   */
+  loading?: boolean,
+  /**
+   * Callback, called by clicking on **close** button. If specified, button will be added automatically
+   * @example <FileHandler onClose={() => doSomething()}>
+   *            text
+   *          </FileHandler>
+   */
+  onClose?: (SyntheticMouseEvent<HTMLDivElement>) => mixed,
```

Comment:
why is `onClose` there? is it open something that triggers this fn on close?

Reply:
it is onClick in button with 'close' Icon - we did it the same in Label or TopLayer and i think it makes sense, but maybe i'm wrong?

Reply:
I mean to re-think possibilities because for sure you can close toplayer, but closing label can be discussing 😄 for sure you can remove label.

Reply:
so you suggest 'onRemove' prop? or 'onClickCloseButton'? or sth else?

Reply:
`onClose` is fine if we are thinking about a button/icon, not something to close. I wanted to say something